libs/mtesitoo-backend-api/src/main/java/com/mtesitoo/backend/service/ProductDetailResponse.java

```diff
@@ -75,6 +75,18 @@ public Product parseResponse(String response) throws JSONException {
                 currencyCode = jsonProduct.getString("currency_code");
             }
 
+            int numPendingOrders = 0, numProcessingOrders = 0;
+            if (jsonProduct.get("order_counts") instanceof JSONObject) {
+                JSONObject jsonObject = jsonProduct.getJSONObject("order_counts");

```

Comment:
Would this PR need to work with already created products? I have 2 products that were created before adding this fields, so the app fails to parse the product, skipping this code. This ends up in an empty product screen

Reply:
i've wrote this code using pre-existing products only. Haven't created a new product before writing this piece. Technically, 'Orders Pending' in ProductList screen should be empty/zero, when there's no such information available. From parsing point of view, "order_counts" will be accessed only when its available. It shouldn't really impact any past or future products. What are you seeing ? Do you some strange behavior in logcat ?

Reply:
The problem is that my products doesn't have any "order_counts" key, resulting in a parsing exception. Because of that, after selecting a product in the list to see its details, the product detail screen is just empty, presenting the toast "Error getting product images"
<img width="321" alt="screen shot 2017-10-09 at 12 13 51" src="https://user-images.githubusercontent.com/1443913/31335664-6c7fe3d6-aceb-11e7-8d4d-3ba8fb7fe497.png">
<img width="312" alt="screen shot 2017-10-09 at 12 14 07" src="https://user-images.githubusercontent.com/1443913/31335665-6c85f938-aceb-11e7-8c29-522074ffc1ab.png">

Reply:
@edu11 : Doesn't look like a parsing exception on `order_counts`. I see `Unexpected response code 404 for https://tesitoo.com/api/v1//product/product/190` in logcat, which explains error message above 'Error getting product images'. One interesting thing I noticed was that above error is only shown to products with status as 'Disable' (even in `master` branch). And same behavior I can reproduce on `master` branch as well (where there's no parsing code for `order_counts` available yet). I would recommend to test out this PR against a test user that we use 'vendor1/password' for scope of this PR code changes. Let me know if you see above issue with that user. We need to figure out if the above issue ***was caught*** as part of this PR or ***was caused*** by this PR. To me it looks like it's caught here (since I can repro same error in master), but would want someone else also run in their environment.

Reply:
I'm probably not completely aware of the implications of this PR and the changes that has been done to products, but during my test, the error fired was definitely related to the parsing of the response:

<img width="1008" alt="screen shot 2017-10-09 at 22 42 33" src="https://user-images.githubusercontent.com/1443913/31359906-eb6c44a2-ad43-11e7-9eef-6e7b289c73e9.png">

No matter the user I introduce (mine or vendor1), I get the same error. Pulling your latest changes, allows me to view product details without a problem, so as for the error I experienced, I'd mark it as resolve now.

Regarding 'disable' products error, I think you may have found a different error here. However, I don't see any product with status as 'Disable', just 'Enabled' or 'Pending', not even for vendor1, so I'd need some background to be able to reproduce it, but it does look like an issue that's been caught here, not cause by this PR